src/agent/utils/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use anyhow::Result;
+use async_trait::async_trait;
+use backoff::{self, future::FutureOperation, ExponentialBackoff};
+use reqwest::Response;
+use std::time::Duration;
+
+const DEFAULT_RETRY_PERIOD: Duration = Duration::from_millis(500);
+const MAX_ELAPSED_TIME: Duration = Duration::from_secs(60 * 5);
```

Comment:
While reviewing this, I found [Azure retry guidance](https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines).

We probably want to tune this to align with the conservative side of Azure retry guidance.

* Max of 5 total tries (4 retries plus the initial)
* Max elapsed duration of 30s
* Support error code specific knowledge to enable retries (only retry if it's a connection issue OR one of these HTTP error codes)
    * 408 Request Timeout
    * 429 Too Many Requests
    * 500 Internal Server Error 
    * 502 Bad Gateway
    * 503 Service Unavailable
    * 504 Gateway Timeout